packages/@aws-cdk/aws-logs/test/log-retention.test.ts

```diff
@@ -82,6 +159,33 @@ describe('log retention', () => {
             'Effect': 'Allow',
             'Resource': '*',
           },
+          {
```

Comment:
We need test cases to check that this isn't added when the removal policy is `RETAIN` or when the user doesn't set a removal policy.